app/page.tsx

```diff
@@ -1,75 +1,99 @@
 'use client';
 
 import React from 'react';
-import { useForm, Controller } from 'react-hook-form';
-import { Select, MenuItem, TextField, FormControl, InputLabel, Button, Box, Checkbox, FormControlLabel } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Divider from '@mui/material/Divider';
+import TextInput from '@/components/TextInput';
+import { FormValues } from '@/types';
 
-type FormValues = {
-  firstName: string;
-  lastName: string;
-  email: string;
-  phone: string;
-  whiskey: string;
-  quantity: number;
-  address: string;
-  ageVerified: boolean;
-};
+const whiskeyOptions = ['Whiskey Option 1', 'Whiskey Option 2', 'Whiskey Option 3'];
 
-export default function WhiskeyPurchaseForm() {
-  const { register, handleSubmit, control } = useForm<FormValues>({
+const Page = () => {
+  const { handleSubmit, control } = useForm<FormValues>({
```

Comment:
Note: we don't want to pass `defaultValues` to `useForm`. We let the form initialize by itself. If TypyScript is complaining about missing values, let me know.

Reply:
everything is working fine, I just get this warning in the console when I start to fill the textinput fields

`Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. `

Reply:
You need to account for that. Your `field` object inside `TextInput` has a `value` property. Right now you spread the `field` onto `TextInput` which results in this component accepting whatever value is passed by `value` property. You need to make sure that if is present, then you pass it as is. Else (if non-existent, like null), then you pass an empty string.